pkg/route/validation/validation_test.go

```diff
@@ -1946,12 +2020,162 @@ func TestValidateTLS(t *testing.T) {
+				},
+			},
+			opts:           routecommon.RouteValidationOptions{AllowExternalCertificates: true},
+			expectedErrors: 1,
+		},
+		{
+			name: "Invalid Passthrough termination as externalCertificate and certificate set",
+			route: &routev1.Route{
+				Spec: routev1.RouteSpec{
+					TLS: &routev1.TLSConfig{
+						Termination: routev1.TLSTerminationPassthrough,
+						ExternalCertificate: &routev1.LocalObjectReference{
+							Name: "tls-secret",
+						},
+						Certificate: "dummy",
+					},
+				},
+			},
+			opts:           routecommon.RouteValidationOptions{AllowExternalCertificates: true},
+			expectedErrors: 2,
+		},
+		{
+			name: "Invalid Edge termination as externalCertificate and certificate",
+			route: &routev1.Route{
+				Spec: routev1.RouteSpec{
+					TLS: &routev1.TLSConfig{
+						Termination: routev1.TLSTerminationEdge,
+						ExternalCertificate: &routev1.LocalObjectReference{
+							Name: "tls-secret",
+						},
+						Certificate: "dummy",
+					},
+				},
+			},
+			opts:           routecommon.RouteValidationOptions{AllowExternalCertificates: true},
+			expectedErrors: 1,
+		},
+		{
+			name: "Invalid externalCertificate as not authorized",
+			route: &routev1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "route-test",
+					Namespace: "sandbox",
+				},
+				Spec: routev1.RouteSpec{
+					TLS: &routev1.TLSConfig{
+						Termination: routev1.TLSTerminationEdge,
+						ExternalCertificate: &routev1.LocalObjectReference{
+							Name: "tls-secret",
+						},
+					},
+				},
+			},
+			allow:          false,
+			validType:      true,
+			validNS:        true,
+			opts:           routecommon.RouteValidationOptions{AllowExternalCertificates: true},
+			expectedErrors: 5,
+		},
+		{
+			name: "Invalid externalCertificate as secret not found",
+			route: &routev1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "route-test",
+					Namespace: "sandbox",
+				},
+				Spec: routev1.RouteSpec{
+					TLS: &routev1.TLSConfig{
+						Termination: routev1.TLSTerminationEdge,
+						ExternalCertificate: &routev1.LocalObjectReference{
+							Name: "tls-secret",
+						},
+					},
+				},
+			},
+			allow:          true,
+			validNS:        false,
+			validType:      true,
+			opts:           routecommon.RouteValidationOptions{AllowExternalCertificates: true},
+			expectedErrors: 1,
+		},
+		{
+			name: "Invalid externalCertificate as secret of incorrect type",
+			route: &routev1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "route-test",
+					Namespace: "sandbox",
+				},
+				Spec: routev1.RouteSpec{
+					TLS: &routev1.TLSConfig{
+						Termination: routev1.TLSTerminationEdge,
+						ExternalCertificate: &routev1.LocalObjectReference{
+							Name: "tls-secret",
+						},
+					},
+				},
+			},
+			allow:          true,
+			validNS:        true,
+			validType:      false,
+			opts:           routecommon.RouteValidationOptions{AllowExternalCertificates: true},
 			expectedErrors: 1,
 		},
```

Comment:
I see only error cases for the new code, we need some non error test cases (when `expectedErrors: 0`).

Reply:
Thanks for the catch, added new test (`Valid Edge termination with externalCertificate`)  where `expectedErrors: 0`

Reply:
Where is it? I cannot see it.